pkg/apis/componentconfig/types.go

```diff
@@ -346,3 +346,130 @@ type LeaderElectionConfiguration struct {
+	// horizontalPodAutoscalerSyncPeriod is The period for syncing the number of pods in horizontal pod autoscaler.
+	HorizontalPodAutoscalerSyncPeriod unversioned.Duration `json:"horizontalPodAutoscalerSyncPeriod"`
+	// deploymentControllerSyncPeriod is Period for syncing the deployments.
+	DeploymentControllerSyncPeriod unversioned.Duration `json:"deploymentControllerSyncPeriod"`
+	// podEvictionTimeout is The grace period for deleting pods on failed nodes.
+	PodEvictionTimeout unversioned.Duration `json:"podEvictionTimeout"`
+	// deletingPodsQps is Number of nodes per second on which pods are deleted in case of node failure.
+	DeletingPodsQps float32 `json:"deletingPodsQps"`
+	// deletingPodsBurst is Number of nodes on which pods are bursty deleted in case of node failure. For more details look into RateLimiter.
+	DeletingPodsBurst int `json:"deletingPodsBurst"`
+	// Amount of time which we allow running Node to be unresponsive before
+	// marking it unhealty. Must be N times more than kubelet's nodeStatusUpdateFrequency,
+	// where N means number of retries allowed for kubelet to post node status.
+	NodeMonitorGracePeriod unversioned.Duration `json:"nodeMonitorGracePeriod"`
+	RegisterRetryCount     int                  `json:"registerRetryCount"`
+	// Amount of time which we allow starting Node to be unresponsive before marking it unhealty.
+	NodeStartupGracePeriod unversioned.Duration `json:"nodeStartupGracePeriod"`
+	// The period for syncing NodeStatus in NodeController.
+	NodeMonitorPeriod unversioned.Duration `json:"nodeMonitorPeriod"`
+	// serviceAccountKeyFile is Filename containing a PEM-encoded private RSA key used to sign service account tokens.
+	ServiceAccountKeyFile string `json:"serviceAccountKeyFile"`
+	// enableProfiling is Enable profiling via web interface host:port/debug/pprof/
+	EnableProfiling bool `json:"enableProfiling"`
+	// clusterName is The instance prefix for the cluster
+	ClusterName string `json:"clusterName"`
+	// clusterCIDR is CIDR Range for Pods in cluster.
+	ClusterCIDR string `json:"clusterCIDR"`
+	// allocateNodeCIDRs is Should CIDRs for Pods be allocated and set on the
+	// cloud provider.
+	AllocateNodeCIDRs bool `json:"allocateNodeCIDRs"`
+	// master is the address of the Kubernetes API server (overrides any value
+	// in kubeconfig)
+	RootCAFile string `json:"rootCAFile"`
+	// kubeAPIQPS is QPS to use while talking with kubernetes apiserver
+	KubeAPIQPS float32 `json:"kubeAPIQPS"`
+	// kubeAPIBurst is Burst to use while talking with kubernetes apiserver
+	KubeAPIBurst int `json:"kubeAPIBurst"`
+	// leaderElection defines the configuration of leader election client.
+	LeaderElection LeaderElectionConfiguration `json:"leaderElection"`
+	// vloumeConfiguration
+	VolumeConfiguration VolumeConfiguration `json:"volumeConfiguration"`
+}
+
+// VolumeConfigFlags contains *all* enumerated flags meant to configure all volume
+// plugins. From this config, the binary will create many instances of
```

Comment:
s/the binary/the ControllerManager/binary/

does the instance of volume.VolumeConfig for each plugin contain just the flags/config relevant to that plugin? if so, say that.